sodetlib/operations/bias_wave.py

```diff
@@ -0,0 +1,326 @@
+import time
+import os
+import traceback
+import numpy as np
+import sodetlib as sdl
+import matplotlib.pyplot as plt
+import scipy.optimize
+from sodetlib.operations import bias_steps
+
+np.seterr(all='ignore')
+
+def play_bias_wave(S, bias_group, freqs_wave, amp_wave, duration,
+                   dc_bias=None):
+    """
+    UPDATE THE DOCSTRING
+    """
+    start_times, stop_times = [], []
+
+    if dc_bias is None:
+        dc_bias = S.get_tes_bias_bipolar_array()[bias_group]
+
+    for freq in freqs_wave:
+        S.log(f"BL sine wave with bg={bias_group}, freq={freq}")
+        S.play_sine_tes(bias_group, amp_wave, freq, dc_amp=dc_bias)
+        start_times.append(time.time())
+        time.sleep(duration)
+        stop_times.append(time.time())
+        S.set_rtm_arb_waveform_enable(0)
+        S.set_tes_bias_bipolar(bias_group, dc_bias)
+    return start_times, stop_times
+
+class BiasWaveAnalysis:
+    """
+    UPDATE THE DOCSTRING
+    """
+    def __init__(self, S=None, cfg=None, bgs=None, run_kwargs=None):
+        self._S = S
+        self._cfg = cfg
+
+        self.bgs = bgs
+        self.am = None
+        # WHAT DO I REPLACE THIS WITH??
+        # self.edge_idxs = None
+        
+        # DOES BEN USE THIS?
+        # self.transition_range = None
```

Comment:
I'm not sure what this is so... I probably did not use it before? perhaps you could clarify?

Reply:
Oh ya...sorry I should probably just delete this out. I think it's sort of a vestige from an old version of the bias_step analysis where we would rely on this parameter to calculate the parameters differently depending if the bias step was taken when in the transition or not. Now this is estimated from the data directly and analysis updated accordingly.

Reply:
gotcha gotcha, thanks for clarifying!